media/js/firefox/concerts.js

```diff
@@ -1,6 +1,173 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
 (function(Mozilla) {
     'use strict';
 
+    var ConcertPage = {
+        US_COUNTRY_CODE: 'us',
+        COOKIE_ID: 'firefox-concerts',
+        COOKIE_EXPIRATION_DAYS: 3,
+    };
+
+    var _geoTimeout;
+    var _requestComplete = false;
+
+    // take params from URL and pass through signup button
+    ConcertPage.passThroughParams = function() {
+        var params = window.location.search.slice(1);
+        ['source','medium','campaign','content'].forEach(function(p) {
+            var param = 'utm_' + p;
+            if (params.indexOf(param) >= 0) {
+                var regex = RegExp(param + '=([^\#\&\?]+).*$');
+                if (regex.test(params)) {
+                    var value = params.match(regex)[1];
+                    document.getElementById(param).value = value;
+                }
+            }
+        });
+    };
+
+    ConcertPage.getLocation = function() {
+        // should /country-code.json be slow to load,
+        // just show the regular messaging after 3 seconds waiting.
+        _geoTimeout = setTimeout(ConcertPage.onRequestComplete, 3000);
+
+        var xhr = new window.XMLHttpRequest();
+
+        xhr.onload = function(r) {
+            var country = 'none';
+
+            // make sure status is in the acceptable range
+            if (r.target.status >= 200 && r.target.status < 300) {
+
+                try {
+                    country = JSON.parse(r.target.responseText).country_code.toLowerCase();
+                } catch (e) {
+                    country = 'none';
+                }
+            }
+
+            ConcertPage.onRequestComplete(country);
+        };
+
+        xhr.open('GET', '/country-code.json');
+        // must come after open call above for IE 10 & 11
+        xhr.timeout = 2000;
+        xhr.send();
+    };
+
+    ConcertPage.hasGeoOverride = function(location) {
+        var loc = location || window.location.search;
+        if (loc.indexOf('geo=') !== -1) {
+            var urlRe = /geo=([a-z]{2})/i;
+            var match = urlRe.exec(loc);
+            if (match) {
+                return match[1].toLowerCase();
+            }
+            return false;
+        }
+        return false;
+    };
+
+    ConcertPage.verifyLocation = function(location) {
+        if (location) {
+            return location !== ConcertPage.US_COUNTRY_CODE;
+        }
+
+        return false;
+    };
+
+    ConcertPage.onRequestComplete = function(data) {
+        var country = typeof data === 'string' ? data : 'none';
+
+        clearTimeout(_geoTimeout);
+
+        if (!_requestComplete) {
+            _requestComplete = true;
+
+            // Set a cookie so we don't have to query location on repeated page loads.
```

Comment:
When I set my cookie to be `ca` the page still loads.

I know we can't really test the geofencing until it's on stage though :(